From 8f76723d26334f06297f26f88d560eddafb607cf Mon Sep 17 00:00:00 2001
From: Sam Nazarko <email@samnazarko.co.uk>
Date: Mon, 20 Mar 2017 02:45:58 +0000
Subject: [PATCH] Ensure that add-ons can run before shutdown / reboot. This
 allows correct teardown when rebooting or shutting down from the Profile
 screen before login when add-ons are still being suppressed

Signed-off-by: Sam Nazarko <email@samnazarko.co.uk>
---
 xbmc/application/ApplicationMessageHandling.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/xbmc/application/ApplicationMessageHandling.cpp b/xbmc/application/ApplicationMessageHandling.cpp
index eeb099e6ae..8a76c229cb 100644
--- a/xbmc/application/ApplicationMessageHandling.cpp
+++ b/xbmc/application/ApplicationMessageHandling.cpp
@@ -119,6 +119,7 @@ void CApplicationMessageHandling::OnApplicationMessage(MESSAGING::ThreadMessage*
   {
     case TMSG_POWERDOWN:
       if (m_app.Stop(EXITCODE_POWERDOWN))
+        CServiceBroker::GetAppMessenger()->PostMsg(TMSG_SETOSMCWALKTHROUGHSTATE, 2);
         CServiceBroker::GetPowerManager().Powerdown();
       break;
 
@@ -134,6 +135,7 @@ void CApplicationMessageHandling::OnApplicationMessage(MESSAGING::ThreadMessage*
     case TMSG_RESTART:
     case TMSG_RESET:
       if (m_app.Stop(EXITCODE_REBOOT))
+        CServiceBroker::GetAppMessenger()->PostMsg(TMSG_SETOSMCWALKTHROUGHSTATE, 2);
         CServiceBroker::GetPowerManager().Reboot();
       break;
 
-- 
2.34.1

